python-avd/pyavd/_eos_designs/shared_utils/mlag.py

```diff
@@ -186,3 +186,16 @@ def mlag_peer_ibgp_ip(self: SharedUtils) -> str:
             return self.mlag_peer_l3_ip
 
         return self.mlag_peer_ip
+
+    @cached_property
+    def use_separate_peer_group_for_mlag_vrfs(self: SharedUtils) -> bool:
+        return (
+            self.inputs.bgp_peer_groups.mlag_ipv4_vrfs_peer
+            and self.inputs.bgp_peer_groups.mlag_ipv4_vrfs_peer.name != self.inputs.bgp_peer_groups.mlag_ipv4_underlay_peer.name
+        )
+
+    @cached_property
+    def mlag_vrfs_peer_group_name(self: SharedUtils) -> dict:
```

Comment:
Please make sure to enable type checking in your dev environment. You should see an error on this type hint.

```suggestion
    def mlag_vrfs_peer_group_name(self: SharedUtils) -> str:
```

Reply:
Sure Claus, I have changed the setting to "strict" now.